lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3699,6 +3698,18 @@ OwningModuleRef circt::firrtl::importFIRFile(SourceMgr &sourceMgr,
           .parseCircuit(annotationsBufs, omirBufs))
     return nullptr;
 
+  // Apply `foldWhenEncodedVerifOp` to all PrintFOp as a post processing.
+  mlir::parallelForEach(
+      context, module.get().getBody()->getOperations(), [&](Operation &op) {
+        // It's dangerous to modify IR in the walk so first accumulate into
+        // buffer.
+        SmallVector<PrintFOp> buffer;
+        op.walk([&buffer](PrintFOp printFOp) { buffer.push_back(printFOp); });
+
+        llvm::for_each(buffer,
+                       [](auto printFOp) { foldWhenEncodedVerifOp(printFOp); });
+      });
```

Comment:
Instead of handling this after the whole file is parsed, can you do this in parseModuleBody?  This will have better cache locality (since the IR just got built) and eliminate the need for another parallel for loop

Reply:
That makes sense. Thanks!